xbmc/windowing/osx/WinSystemIOS.mm

```diff
@@ -346,26 +367,56 @@
     (*i)->OnAppFocusChange(focus);
 }
 
-void CWinSystemIOS::VblankHandler(int64_t nowtime, double fps)
+//--------------------------------------------------------------
+//-------------------DisplayLink stuff
+@implementation IOSDisplayLinkCallback
+@synthesize _videoSyncImpl;
+//--------------------------------------------------------------
+- (void) runDisplayLink;
 {
-  if (m_VideoSync)
-    m_VideoSync->VblankHandler(nowtime, fps);
+  NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+  if (_videoSyncImpl != nil)
```

Comment:
no my assumption was that deinit doesn't return as long as a callback is running - so refclock can't tear down until deinitdisplaylink returns